taiga2/controllers/endpoint.py

```diff
@@ -610,6 +618,67 @@ def create_dataset(sessionDatasetInfo):
     return flask.jsonify(added_dataset.id)
 
 
+def create_new_dataset_version_from_session(
+    current_user,
+    datafile_names,
+    session_id,
+    dataset_id,
+    new_description,
+    changes_description,
+    dataset_version,
+    add_existing_files,
+):
+    models_controller.lock()
+    try:
+        if add_existing_files:
+            added_datafiles = models_controller.get_previous_version_and_added_datafiles(
+                dataset_version, dataset_id, datafile_names, session_id
+            )
+        else:
+            added_datafiles = models_controller.add_datafiles_from_session(session_id)
+
+        all_datafile_ids = [datafile.id for datafile in added_datafiles]
+
+        new_dataset_version = models_controller.add_dataset_version(
+            dataset_id=dataset_id,
+            datafiles_ids=all_datafile_ids,
+            new_description=new_description,
+            changes_description=changes_description,
+        )
+
+    except:
```

Comment:
Does this `except` need to be here? I'd normally expect any exception to be raised and handled at a higher layer. At the same time `api_error("")` looks incomplete. Was this maybe put here for debugging?

Reply:
I'm a little confused about this. I'd like to talk about it when you have time on Tuesday. I added it to our check-in agenda, too

Specifically, I am confused about how to properly handle the try/except. (Sorry about not raising this issue earlier. I only just now realized I was confused as I was trying to implement your suggestion.)

My intention with the try/except was to make sure I rollback the session in the event something fails before the lock can be released with a lock table update.